File: PY/Bryann Valderrama/MinMaxSearch.py
class Pair:
    min = int()
    max = int()


def getMaxMin(lista, n):
    minmax = Pair()
    i = int()
    if (n == 1):
        minmax.max = lista[0]
        minmax.max = lista[0]
        return minmax
    if (lista[0] > lista[1]):
        minmax.max = lista[0]
        minmax.min = lista[1]
    else:
        minmax.max = lista[1]
        minmax.min = lista[0]
    for i in range(2, n):
        if (lista[i] > minmax.max):
            minmax.max = lista[i]
        elif (lista[i] < minmax.min):
            minmax.min = lista[i]
    return minmax


if __name__ == '__main__':
    lista = [4, -1, 6, 23, 7, 78, 2]
    tamanio = int(len(lista))
    resultado = getMaxMin(lista, tamanio)
    print('El minimo es ', resultado.min)
    print('El maximo es ', resultado.max)

File: Java/Bryann Valderrama/MinMaxSearch.java
public class Main {

    public static void main(String[] args) {
        int arreglo[] = {1, 4, 6, 3, 8, 2};
        int tamanio = arreglo.length;
        Pair resultado = getMaxMin(arreglo, tamanio);
        System.out.println("El minimo elemento es " + resultado.min);
        System.out.println("El maximo elemento es " + resultado.max);
    }

    static class Pair {

        int min;
        int max;
    }

    static Pair getMaxMin(int arreglo[], int n) {
        Pair minmax = new Pair();
        int i;
        if (n == 1) {
            minmax.max = arreglo[0];
            minmax.min = arreglo[0];
            return minmax;
        }
        if (arreglo[0] > arreglo[1]) {
            minmax.max = arreglo[0];
            minmax.min = arreglo[1];
        } else {
            minmax.max = arreglo[1];
            minmax.min = arreglo[0];
        }
        for (i = 2; i < n; i++) {
            if (arreglo[i] > minmax.max) {
                minmax.max = arreglo[i];
            } else if (arreglo[i] < minmax.min) {
                minmax.min = arreglo[i];
            }
        }
        return minmax;
    }
}
